File: app/src/main/java/com/apm29/kotlinapp/base/BaseUI.java
package com.apm29.kotlinapp.base;

/**
 * Created by apm29 on 2017/9/7.
 */

public interface BaseUI {
    void startLoading();
    void stopLoading();
    void onError(String error);
    <N> void onNewData(N data);
    void onEmpty();
}

File: app/src/main/java/com/apm29/kotlinapp/base/BaseUI.kt
package com.apm29.kotlinapp.base

import io.reactivex.disposables.CompositeDisposable

/**
 * Created by apm29 on 2017/9/7.
 */

interface BaseUI {
    //请求的订阅链接,在页面销毁时提前取消订阅
    var mDisposables:CompositeDisposable
    fun startLoading()
    fun stopLoading()
    fun onError(error: String?)
    fun onNewData(data: Any?)
    fun onEmpty()
}
